File: 496. Next Greater Element I/solution_naive.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        vector<int> result;
        for (int num1: nums1) {
            auto it = find(nums2.begin(), nums2.end(), num1);
            int greaterElement = -1;
            for (auto it2 = it; it2 < nums2.end(); it2 += 1) {
                if (*it2 > num1) {
                    greaterElement = *it2;
                    break;
                }
            }
            result.push_back(greaterElement);
        }
        
        return result;
    }
};

File: 496. Next Greater Element I/solution_naive.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result = []
        
        for num1 in nums1:
            offset = nums2.index(num1)
            greater_element = -1
            for idx, num2 in enumerate(nums2[offset:]):
                if num2 > num1:
                    greater_element = num2
                    break
            result.append(greater_element)
        
        return result
